hydra-node/src/Hydra/Options.hs

```diff
@@ -88,15 +89,17 @@ instance Arbitrary ParamMismatch where
   arbitrary = genericArbitrary
 
 data Command
-  = Run RunOptions
+  = RunWithArguments RunOptions
+  | RunWithConfig FilePath
   | Publish PublishOptions
   | GenHydraKey GenerateKeyPair
   deriving (Show, Eq)
 
 commandParser :: Parser Command
 commandParser =
   asum
-    [ Run <$> runOptionsParser
+    [ RunWithArguments <$> runOptionsParser
+    , RunWithConfig <$> configFileParser
```

Comment:
What happens if both are given? `-c` and any of the other options?

Reply:
Our default still is `RunWithArguments` and asum would use alternative between actions so it should ignore `-c` in case all other arguments are correct. I could write a test for it so we cover this scenario as well.